src/Schedule/RuleInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace GroupLife\Core\Schedule;
+
+interface RuleInterface
+{
+    public function includedDays(\DateTime $from, \DateInterval $period, array $schedule): array;
+
+    public function excludedDays(\DateTime $from, \DateInterval $period, array $schedule): array;
```

Comment:
Let's start here.

Not clear why this 'array $schedule' parameter is exists. Is it some list of dates? You don't need it to list excluded or included days according to some rule.